File: apps/sdkwork-rtc-pc/packages/sdkwork-rtc-pc-admin-core/src/services/providerWebhookService.ts
import type { AuthTokenManager } from "@sdkwork/sdk-common";

import type { ProviderWebhookEvent } from "../types/providerWebhookEvent";
import { resolveBackendRtcClient, type RtcBackendClientOptions, type RtcBackendClientSource } from "./backendClient";

interface ListResponse {
  items: ProviderWebhookEvent[];
  nextCursor?: string | null;
}

export class ProviderWebhookService {
  private readonly client;

  constructor(
    baseUrlOrClient: RtcBackendClientSource,
    tokenManagerOrOptions?: AuthTokenManager | RtcBackendClientOptions,
  ) {
    this.client = resolveBackendRtcClient(baseUrlOrClient, tokenManagerOrOptions);
  }

  async listEvents(params?: {
    page?: number;
    limit?: number;
    cursor?: string;
    search?: string;
    sort?: string;
  }): Promise<ListResponse> {
    const response = await this.client.rtcProviderWebhooks.rtc.providerWebhooks.events.list({
      page: params?.page,
      pageSize: params?.limit,
      cursor: params?.cursor,
      q: params?.search,
      sort: params?.sort,
    });
    // The generated SDK unwraps the sdkwork-v3 envelope, so the response is
    // already the `data` payload ({ items, pageInfo }).
    const nextCursor = response.pageInfo?.nextCursor;
    return {
      items: response.items,
      nextCursor: nextCursor && nextCursor.length > 0 ? nextCursor : null,
    };
  }
}

File: apps/sdkwork-rtc-flutter-mobile/packages/sdkwork_rtc_flutter_mobile_admin_core/lib/src/services/provider_webhook_service.dart
import '../admin_sdk_mapper.dart';
import '../backend_rtc_client.dart';
import '../models/paginated_list_result.dart';
import '../models/provider_webhook_event.dart';

class ProviderWebhookService {
  final SdkworkBackendClient _client;

  ProviderWebhookService(this._client);

  Future<PaginatedListResult<ProviderWebhookEvent>> listEvents({
    int? page,
    int? limit,
    String? cursor,
    String? search,
    String? sort,
  }) async {
    final response = await _client.rtcProviderWebhooks.eventsList(
      page,
      limit,
      cursor,
      sort,
      search,
    );
    return PaginatedListResult(
      items: backendResponseItems(response)
          .map(ProviderWebhookEvent.fromJson)
          .toList(),
      nextCursor: backendResponseNextCursor(response),
    );
  }
}
